private/model/api/passes.go

```diff
@@ -244,6 +244,46 @@ func (a *API) renameExportable() {
 	}
 }
 
+// renameCollidingFields will rename any fields that uses an SDK or Golang
+// specific name.
+func (a *API) renameCollidingFields() {
+	for _, v := range a.Shapes {
+		namesWithSet := map[string]struct{}{}
+		for k, field := range v.MemberRefs {
+			if strings.HasPrefix(k, "Set") {
+				namesWithSet[k] = struct{}{}
+			}
+
+			if collides(k) {
+				delete(v.MemberRefs, k)
+				v.MemberRefs[k+"_"] = field
+			}
+		}
+
+		// checks if any field names collide with setters.
+		for name := range namesWithSet {
+			setterName := fmt.Sprintf("Set%s", name)
```

Comment:
Would be good to move this specific rename logic within the this for loop into its own function.  Then the logic can be made more generic to just rename colliding reference, instead of duplicated logic for Set prefix, and Colliding reference name.